apps/omg_db/lib/db.ex

```diff
@@ -122,6 +110,19 @@ defmodule OMG.DB do
   def get_single_value(parameter_name), do: driver().get_single_value(parameter_name)
   def get_single_value(parameter_name, server), do: driver().get_single_value(parameter_name, server)
 
+  @doc """
+  This is generic DB function that can get the specific data of a specific type.
+  If it is a single value data, use get_single_value/1 instead.
+  """
+  def get(type, specific_key), do: driver().get(type, specific_key)
+  def get(type, specific_key, server), do: driver().get(type, specific_key, server)
+
+  @doc """
+  This is generic DB function that can get all data of a specific type.
+  """
+  def get_all_by_type(type), do: driver().get_all_by_type(type)
```

Comment:
Prefix search in RocksDB is faster comparing to the LevelDB however we found it still too slow for fetching UTXOs set. We could extend the time we're waiting for the response - but it does not helped as much as it blocks other servers to get the data (it blocks on `OMG.DB.RocksDB.Server` GenServer)

So exposing generic function could bring as troubles.

Reply:
well....that is how it is used in current implementation no matter extra explosion or not. It just seems obvious that as a KV store, client of DB need to access all data of a type from time to time. It is used in `utxos()`, `exit_infos()`, `in_flight_exits_info()`, and `competitors_info()`

But let me try to explain my motivation of expose the function. If we simplify the `OMG.DB` behavior to only simple generic functions like: `get`, `get_multiple`, `write_multiple` (not exactly this function name, but the actions), it will allow us to swap the DB with easier implementation (few functions only). The DAO-ish model module has the potential to not need to make any change at all if it already has a way to inject to different DB server/driver it is using.